### PR TITLE
♻️Iframe Refactor

### PR DESCRIPTION
**Pull request by calebcordry**
- Renders outer iframe only once 
- Renders using srcdoc if supported, doc write if not.
- Deletes utils/document-html.js

Should fix relative url problem. Check out the netlifly, way less flashing 🎉

@alanorozco LMK if you want to keep the utils around and I will revert the commit.

### CODE REVIEW

File: src/utils/iframe.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * @return {!Promise}
+ */
+export function whenIframeLoaded(iframe) {
+  const {promise, resolve} = new Deferred();
+  iframe.addEventListener(
+    'load',
+    () => {
+      resolve(iframe);
+    },
+    {once: true}
```
Comment (alanorozco) [inline src/utils/iframe.js]: 😮 TIL

does this cleanup as well?

File: src/amp-story-ad-preview.js [hunk 2]
```diff
@@ -76,17 +82,43 @@ export default class AmpStoryAdPreview {
...
+    this.determineIframeStrategy_();
+
     // We're already keeping this value, so let's just remove the attribute to
     // clear memory.
     element.removeAttribute('data-template');
 
-    render(WrappedIframe(), this.element);
+    render(
+      WrappedIframe({
+        storyTemplate: this.storyTemplate_,
```
Comment (alanorozco) [inline src/amp-story-ad-preview.js]: Maybe just pass a `srcdoc` attr directly? and `this.useSourcedoc_ ? this.storyTemplate_ : ''`. Restricts logic to one place.

**Comment by calebcordry** [comment]
Eh, kind of. `amp-ad` does a lot of stuff that we don't need to worry about. Also it doesn't need to worry about loadevents/lit etc.

FriendlyIframeEmbed does the srcdoc but fallback to document write though.

File: src/utils/iframe.js (hunk 1 above)
Comment (calebcordry) [reply src/utils/iframe.js]: Yep :)

File: src/amp-story-ad-preview.js (hunk 2 above)
Comment (calebcordry) [reply src/amp-story-ad-preview.js]: Done.

File: src/utils/iframe.js (hunk 1 above)
Comment (alanorozco) [reply src/utils/iframe.js]: wonderful

**Comment by alanorozco** [comment]
Just checked out the deploy preview, *way* nicer.

**Pull request merged**